packages/vscode-extension/src/debug/teamsfxTaskProvider.ts

```diff
@@ -28,8 +31,27 @@ export class TeamsfxTaskProvider implements vscode.TaskProvider {
         return tasks;
       }
 
-      const programmingLanguage = await commonUtils.getProgrammingLanguage();
-      const localEnv = await commonUtils.getLocalEnv();
+      const localEnvManager = new LocalEnvManager(VsCodeLogInstance, ExtTelemetry.reporter);
+      let projectSettings: ProjectSettings;
+      let localSettings: Json | undefined;
+      let localEnv: { [key: string]: string } | undefined;
+
+      try {
+        projectSettings = await localEnvManager.getProjectSettings(workspacePath);
+        localSettings = await localEnvManager.getLocalSettings(workspacePath, {
+          projectId: projectSettings.projectId,
+        });
+        localEnv = await localEnvManager.getLocalDebugEnvs(
+          workspacePath,
+          projectSettings,
+          localSettings
+        );
+      } catch (error: any) {
+        await showError(error);
```

Comment:
I remember `await showError` will block the thread since in `showError` it uses `await vscode.window.showErrorMessage`.

Reply:
Right... removed `await`.